### PR TITLE
Demo/gradle

### PR DESCRIPTION
**Pull request by ZenHarbinger**
Add demo for using the gradle plugin.

### CODE REVIEW

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

**Comment by sergiusens** [comment]
Nice

You can get rid of the wrapper and just put that in the command entry replacing bin/wrapper

**Comment by ZenHarbinger** [comment]
OK, I've removed the wrapper file and Makefile (and associated steps in the snapcraft.yaml file).
Tested on my local machine and works.

**Comment by kyrofa** [comment]
CLA verified.

**Comment by sergiusens** [comment]
ok to test

**Comment by sergiusens** [comment]
looks good, but what is `demos/gradle/gradle/wrapper/gradle-wrapper.jar` ?

**Comment by ZenHarbinger** [comment]
It's a wrapper jar which will download the appropriate Gradle version to do the gradle work.  All gradlew projects have this when using the gradlew wrapper script.
This means that gradle itself does not need to be installed to run a gradlew script.

**Comment by sergiusens** [comment]
Ok, I see now `CLASSPATH=$APP_HOME/gradle/wrapper/gradle-wrapper.jar`. This makes me a bit nervous though as it is a binary in our source.

Let me think about this for a bit.

File: demos/gradle/gradlew [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+fi
+
+# Split up the JVM_OPTS And GRADLE_OPTS values into an array, following the shell quoting and substitution rules
+function splitJvmOpts() {
+    JVM_OPTS=("$@")
+}
+eval splitJvmOpts $DEFAULT_JVM_OPTS $JAVA_OPTS $GRADLE_OPTS
+JVM_OPTS[${#JVM_OPTS[*]}]="-Dorg.gradle.appname=$APP_BASE_NAME"
+
+exec "$JAVACMD" "${JVM_OPTS[@]}" -classpath "$CLASSPATH" org.gradle.wrapper.GradleWrapperMain "$@"
```
Comment (elopio) [inline demos/gradle/gradlew]: we don't need the darwin and cywin sections, right?

**Comment by ZenHarbinger** [comment]
That script is standard from gradle.  Developers are not usually going to edit/change/create this file.

**Comment by sergiusens** [comment]
Do you mind updating this branch? I think this fell through the cracks and I am so sorry about that.

**Comment by sergiusens** [comment]
@elopio can you help out with adding the test?

**Comment by elopio** [comment]
Yes, on my to-do.

**Comment by elopio** [comment]
Please review #912.
Thanks for the demo @ZenHarbinger!

**Pull request closed without merging**